test/libsolidity/InlineAssembly.cpp

```diff
@@ -182,6 +182,18 @@ BOOST_AUTO_TEST_CASE(error_tag)
 	BOOST_CHECK(successAssemble("{ invalidJumpLabel }"));
 }
 
+BOOST_AUTO_TEST_CASE(inline_assembly_shadowed_instruction_declaration)
+{
+	// Error message: "Cannot use instruction names for identifier names."
+	BOOST_CHECK(!successAssemble("{ let gas := 1 }"));
+}
+
+BOOST_AUTO_TEST_CASE(inline_assembly_shadowed_instruction_assignment)
+{
+	// Error message: "Cannot use instruction names for identifier names."
+	BOOST_CHECK(!successAssemble("{ 2 =: gas }"));
```

Comment:
`2 =: gas` is still an assignment of 2 to `gas`, just mixing the functional and the non-functional notation as:

`2` is pushed to stack`
`=: gas` pops item off the stack into variable `gas`